components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -400,8 +400,6 @@ void RewardsServiceImpl::OnPreferenceChanged(const std::string& key) {
     return;
   }
 
-  EnableGreaseLion();
```

Comment:
It would be dangerous to do this in response to a preference change, as this currently ends up unloading and reinstalling the Greaselion extension yet leaves the related content script injected and in a non-functional state. It seemed odd that this preference callback was being called at all, but it's due to `StateMigration::FreshInstall` which sets the three inline tip preferences to `true`.

I think not calling this here is the safest way forward until/unless Greaselion is changed to be a bit smarter about how it handles this scenario. For the time being, we may want to suggest to the user that a browser restart is required when updating these settings.